File: python_projects/readability/readability.py
from cs50 import get_string


def main():
    # prompt the user for text
    text = get_string("Text: ")
    total_letters = letter_counter(text)
    total_words = word_counter(text)
    total_sentences = sentence_counter(text)
    average_letters = get_average_letters(total_letters, total_words)
    average_sentences = get_average_sentences(total_sentences, total_words)
    grade_index = round(0.0588 * average_letters - 0.296 * average_sentences - 15.8)
    
    # get grade 
    if grade_index < 1:
        print("Before Grade 1")
    elif grade_index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade_index}")
        
        
def letter_counter(text):
    count_letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            count_letters += 1
    return count_letters
    

def word_counter(text):
    count_words = 0
    for i in range(len(text)):
        if text[i].isspace():
            count_words += 1
    count_words += 1
    return count_words


def sentence_counter(text):
    count_sentences = 0
    for i in range(len(text)):
        if text[i] in ('.', '!', '?'):
            count_sentences += 1
    return count_sentences
    

def get_average_letters(total_letters, total_words):
    return (total_letters / float(total_words)) * 100
    

def get_average_sentences(total_sentences, total_words):
    return float((total_sentences / total_words)) * 100


if __name__ == "__main__":
    main()

File: c_projects/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//prototype
int letter_counter(string);
int word_counter(string);
int sentence_counter(string);

int main(void)
{
    //Get text from user
    string text = get_string("Text: ");
    
    //count letters
    int countLetters = letter_counter(text);
    
    //count words
    int countWords = word_counter(text);
    
    //count sentences
    int countSentences = sentence_counter(text);
    
    //calc avg letters
    float L = (countLetters / (float) countWords) * 100;
    
    //calculate avg sentences
    float S = (countSentences / (float) countWords) * 100;
    
    //find index
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    //print grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
    
    // printf("Text %s\n", text);
    // printf("%i letter(s)\n", countLetters);
    // printf("%i word(s)\n", countWords);
    // printf("%i sentence(s)\n", countSentences);
    //printf("Grade %i\n", index);
    
}


int letter_counter(string letters)
{
    int numLetters = 0;
    for (int i = 0, n = strlen(letters); i < n; i++)
    {
        if (isalpha(letters[i]))
        {
            numLetters++;
        }
    }
    
    return numLetters;
}

int word_counter(string words)
{
    int numWords = 0;
    for (int i = 0, n = strlen(words); i < n; i++)
    {
        if (isspace(words[i]))
        {
            numWords++;
        }
    }
    
    numWords++;
    
    return numWords;
    
}

int sentence_counter(string sentences)
{
    int numSentences = 0;
    
    for (int i = 0, n = strlen(sentences); i < n; i++)
    {
        if (sentences[i] == '.' || sentences[i] == '!' || sentences[i] == '?')
        {
            numSentences++;
        }
    }
    
    return numSentences;
}

